### PR TITLE
Add proposal For CloudCore HA

### LINKED ISSUE #1560: HA support of CloudCore
**What would you like to be added/modified**:

CloudCore is not deployed as a single instance. Ideally it should be HA. So we may do it in two steps:
1. Hot standby
2. Deploy a cluster of CloudCore 

To be implemented for alpha in v1.3

- [x] Add leader election in CloudCore
- [ ] Add healthcheck method to CloudCore for integration with LB, to tell which instance is currently active
- [ ] Add LB for multiple instances

To be implemented for beta
- [ ] Make CloudCore Stateless, ref: https://github.com/kubeedge/kubeedge/issues/1444

**Why is this needed**:

### PR DESCRIPTION
**Pull request by GsssC**
**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
Add Proposal For CloudCore High Availability.
**Which issue(s) this PR fixes**:

Ref #1560  #1444

**Special notes for your reviewer**:

```release-note

```

### CODE REVIEW

**Comment by GsssC** [comment]
@kevin-wangzefeng @fisherxu It has been updated.

File: docs/proposals/cloudcore-ha-design.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  - "@GsssC"
+  - "@Poor12"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@fisherxu"
+  - "@kadisi"
+
+creation-date: 2020-03-28
+last-updated: 2020-05-09
+status: alpha
```
Comment (fisherxu) [inline docs/proposals/cloudcore-ha-design.md:13]: Implemented

File: docs/proposals/cloudcore-ha-design.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  - "@GsssC"
+  - "@Poor12"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@fisherxu"
+  - "@kadisi"
+
+creation-date: 2020-03-28
+last-updated: 2020-05-09
+status: alpha implemented
```
Comment (fisherxu) [inline docs/proposals/cloudcore-ha-design.md:13]: We can delete the alpha  status here, only leave `Implemented`

Comment (GsssC) [reply docs/proposals/cloudcore-ha-design.md]: done : )

**Comment by fisherxu** [comment]
/lgtm
Thanks for the effort :)

**Pull request merged**